lib/OpenQA/Task/Bug/Limit.pm

```diff
@@ -0,0 +1,49 @@
+# Copyright (C) 2020 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Task::Bug::Limit;
+use Mojo::Base 'Mojolicious::Plugin';
+
+sub register {
+    my ($self, $app) = @_;
+    $app->minion->add_task(limit_bugs => \&_limit);
+}
+
+sub _limit {
+    my $job = shift;
+    my $app = $job->app;
+
+    # prevent multiple limit_bugs tasks to run in parallel
+    return $job->finish('Previous limit_bugs job is still active')
+      unless my $guard = $app->minion->guard('limit_bugs_task', 86400);
+
+    # prevent multiple limit_* tasks to run in parallel
+    return $job->retry({delay => 60})
+      unless my $limit_guard = $app->minion->guard('limit_tasks', 86400);
+
+    # cleanup entries in the bug table that are not referenced from any comments
+    my $bugrefs = $app->schema->resultset('Comments')->compute_bugref_count();
+    my %cleaned;
+    for my $bug ($app->schema->resultset('Bugs')->all) {
+        if (!defined $bugrefs->{$bug->bugid}) {
+            $bug->delete;
+            $cleaned{$bug->id} = $bug->bugid;
+        }
+    }
```

Comment:
Since `compute_bugref_count` is only used once, i wonder if this whole block should maybe be moved into a resultset method (`my @bugids = $app->schema->resultset('Comments')->cleanup_bugrefs`?).